reference/mongodb/mongodb/driver/manager/construct.xml

```diff
@@ -11,9 +11,9 @@
   &reftitle.description;
   <methodsynopsis>
    <modifier>final</modifier> <modifier>public</modifier> <methodname>MongoDB\Driver\Manager::__construct</methodname>
-   <methodparam choice="opt"><type>string</type><parameter>uri</parameter><initializer>"mongodb://127.0.0.1/"</initializer></methodparam>
-   <methodparam choice="opt"><type>array</type><parameter>uriOptions</parameter><initializer>array()</initializer></methodparam>
-   <methodparam choice="opt"><type>array</type><parameter>driverOptions</parameter><initializer>array()</initializer></methodparam>
+   <methodparam choice="opt"><type class="union"><type>string</type><type>null</type></type><parameter>uri</parameter><initializer>&null;</initializer></methodparam>
```

Comment:
I realize the Manager stub specifies `null` as a default, but we actually default to "mongodb://127.0.0.1/" in the constructor. That value is defined as `PHONGO_MANAGER_URI_DEFAULT` in `Manager.c`.

Since the stub can refer to a C constant, would it make sense to define that before including `Manager_arginfo.h` and reference it as a `@cvalue`, or does that only work for doc blocks on class constants?

I know we talked about keeping the PHP.net docs exactly consistent with the stubs in anticipation of one day generating these docs _from_ the stubs. If that's the case, perhaps we should just hardcode the default value of `mongodb://127.0.0.1/` in the stub. In that case, we might be able to do away with the `PHONGO_MANAGER_URI_DEFAULT` constant if the arginfo can supply the default value. If not, then we can just leave a comment in the stub and above the constant noting that both values should be kept in sync.

Reply:
I changed the default value to `null`, as passing an explicit `null` does the same thing as omitting it: we use the default connection string instead.

Looking through the arginfo generation file I believe `@cvalue` is not available for default values in method parameters, so we'd have to explicitly use the default connection string anyways. However, even then we should probably document what happens if a user chooses to pass `null` to the connection string, so we may as well use `null` as default value and always document that passing `null` as connection string defaults to using `mongodb://127.0.0.1`.

---

As an aside, while testing generating class synopsis from stub files I noticed that `@cvalue` is not correctly moved over to the synopsis, as the script doesn't have any knowledge of the underlying C code. This is something we'll have to address separately once we decide to use the script for that purpose.